### PR TITLE
CrashReporter: Add "Inspect in HackStudio" button

### PR DESCRIPTION
**Pull request by itamar8910**
Clicking the "Inspect in HackStudio" button opens HackStudio and allows the user to inspect the coredump in the Debug tab.

To support inspecting coredumps in the Debug tab, HackStudio's `DebugInfoWidget` now uses an abstract base class called `ProcessInspector` to display the backtrace instead of interacting directly with the `Debugger` object.
Both `DebugSession` and `Coredump::Inspector` are now derived from `ProcessInspector`.

Screenshots:
crashreporter-coredump

Inspecting the coredump in HackStudio:
hs-coredump

### CODE REVIEW

File: Userland/Applications/CrashReporter/CrashReporterWindow.gml [hunk 1]
```diff
@@ -82,6 +82,13 @@
 
         layout: @GUI::HorizontalBoxLayout
 
+        @GUI::Button {
+            name: "inspect_button"
+            text: "Inspect in HackStudio"
```
Comment (linusg) [inline Userland/Applications/CrashReporter/CrashReporterWindow.gml:87]:
```suggestion
            text: "Inspect in Hack Studio"
```

https://github.com/SerenityOS/serenity/blob/4b34a1302bf748e45fc17febdbf71088e45e8b5b/Base/res/apps/HackStudio.af#L2

File: Userland/DevTools/HackStudio/main.cpp [hunk 2]
```diff
@@ -76,9 +78,12 @@ int main(int argc, char** argv)
     };
 
     window->show();
-
     s_hack_studio_widget->update_actions();
 
+    if (mode_coredump) {
+        s_hack_studio_widget->open_coredump(argument_absolute_path);
+    }
```
Comment (linusg) [inline Userland/DevTools/HackStudio/main.cpp:85]: no curly braces needed for single-line if body

```suggestion
    if (mode_coredump)
        s_hack_studio_widget->open_coredump(argument_absolute_path);
```

File: Userland/Libraries/LibCoredump/Inspector.cpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+void Inspector::parse_loaded_libraries(Function<void(float)> on_progress)
+{
+    size_t number_of_libraries = number_of_libraries_in_coredump();
+    size_t library_index = 0;
+
+    m_reader->for_each_library([this, number_of_libraries, &library_index, &on_progress](Coredump::Reader::LibraryInfo library) {
+        ++library_index;
+        if (on_progress) {
+            on_progress(library_index / (float)number_of_libraries);
+        }
```
Comment (linusg) [inline Userland/Libraries/LibCoredump/Inspector.cpp:43]:
```suggestion
        if (on_progress)
            on_progress(library_index / (float)number_of_libraries);
```

File: Userland/Libraries/LibCoredump/Reader.cpp [hunk 4]
```diff
@@ -292,4 +293,26 @@ const Reader::LibraryData* Reader::library_containing(FlatPtr address) const
     return lib_data;
 }
 
+void Reader::for_each_library(Function<void(LibraryInfo)> func) const
+{
+
```
Comment (linusg) [inline Userland/Libraries/LibCoredump/Reader.cpp:298]: Please don't add useless newlines at the start of a function/if/while/etc body

File: Userland/Libraries/LibCoredump/Reader.h [hunk 5]
```diff
@@ -20,12 +20,20 @@ class Reader {
     AK_MAKE_NONMOVABLE(Reader);
 
 public:
-    static OwnPtr<Reader> create(const String&);
+    static OwnPtr<Reader> create(const StringView&);
```
Comment (linusg) [inline Userland/Libraries/LibCoredump/Reader.h:23]: There's no need to pass StringViews by reference - see 8b1108e4858f797c9216dc8ae4a3918ad50c73b4

```suggestion
    static OwnPtr<Reader> create(StringView);
```

**Review by linusg — CHANGES_REQUESTED** [review]

File: Userland/Applications/CrashReporter/CrashReporterWindow.gml (hunk 1 above)
Comment (linusg) [reply Userland/Applications/CrashReporter/CrashReporterWindow.gml:87]: Should then also be reflected in the commit message :^)

**Comment by itamar8910** [comment]
Done, thanks for reviewing this @linusg :)

**Review by linusg — APPROVED** [review]

**Pull request merged**